xcode/X16/iOS/MessageObserver.m

```diff
@@ -0,0 +1,58 @@
+//
+//  MessageObserver.m
+//  CommanderX16
+//
+//  Created by Pearce, Matthew (Senior Developer) on 04/01/2020.
+//
+
+#import "MessageObserver.h"
+#import "SettingsViewController.h"
+#import "SDL.h"
+#import "SDL_uikitappdelegate.h"
+
+@interface MessageObserver ()
+@property (nonatomic, retain) SettingsViewController *settings;
+@end
+
+@implementation MessageObserver
+
+@synthesize settings;
+
+-(id)init {
+    
+    if (self == [super init]) {
+        
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(showSettings) name:@"ShowKeyboard" object:nil];
+        
+    }
+    
+    return self;
+}
+
+-(void)showSettings {
+    
+    if (!settings) {
+        settings = [[SettingsViewController alloc] initWithNibName:@"SettingsViewController" bundle:nil];
+    } else {
+        [[[self rootView] viewWithTag:666] removeFromSuperview];
```

Comment:
I know this is the first ObjC code in this codebase, so who am I to comment about coding style, but I'd *consistently* use dot notation for properties. Below, you say `settings.view.frame`, so you should say `self.rootView` here. And `window.rootViewController` and `delegate.window` below. (But keep `[SDLUIKitDelegate sharedAppDelegate]`, since it's not a property.)

Reply:
Yep - will go through tidying up as I need to change things. Been doing swift for so long now!